truss/templates/control/control/helpers/inference_server_process_controller.py

```diff
@@ -69,7 +71,12 @@ def is_inference_server_running(self) -> bool:
 
     def check_and_recover_inference_server(self):
         if self.inference_server_started() and not self.is_inference_server_running():
-            self._app_logger.warning(
-                "Inference server seems to have crashed, restarting"
-            )
-            self.start()
+            if not INFERENCE_SERVER_FAILED_FILE.exists():
+                self._app_logger.warning(
+                    "Inference server seems to have crashed, restarting"
+                )
+                self.start()
+            else:
+                self._app_logger.warning(
+                    "Inference server volunatrily crashed, not restarting"
```

Comment:
Maybe we should say something along the lines of 
"Inference server unrecoverable. Try patching"